### PR TITLE
Fix sonar reliability issues

### PR DESCRIPTION
**Pull request by tuomas777**
Fix all reported Sonar reliability issues.

- Enforce Bash double-bracket conditionals through ShellCheck.
- Fix importer edge cases, regex performance, and redundant mappings.
- Complete DateRange comparison support with regression tests.
- Fix duplicate HTML IDs, missing image alt text, and invalid email CSS descriptors.

Refs: LINK-2596

## Summary by CodeRabbit

- **Bug Fixes**
  - Event names now correctly prefer Finnish, fall back to Swedish, and return no name when none is available.
  - Event date ranges now support complete ordering comparisons.

- **Accessibility**
  - Added descriptive text to Helsinki logo images.
  - Improved navigation for the keyword-set search section.

- **Visual Improvements**
  - Adjusted font rendering for more consistent typography across email content.

- **Quality Improvements**
  - Expanded automated coverage for event naming and date-range behavior.

### CODE REVIEW

File: events/renderers/docx.py [hunk 1]
```diff
@@ -126,11 +128,14 @@ def __str__(self):
...
+    def _comparison_key(self):
+        return self.start, self.end
+
     def __lt__(self, other):
-        return self.start < other.start
+        return self._comparison_key() < other._comparison_key()
 
     def __eq__(self, other):
-        return self.start == other.start and self.end == other.end
+        return self._comparison_key() == other._comparison_key()
```
[the remark this answers is not in the document]
Comment (tuomas777) [reply events/renderers/docx.py:138]: Overkill for this.

**Review by danipran — APPROVED** [review]
